.github/workflows/tests.yml

```diff
@@ -0,0 +1,128 @@
+name: Tests
+
+on:
+
+  # On which repository actions to trigger the build.
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allow job to be triggered manually.
+  workflow_dispatch:
+
+# Cancel in-progress jobs when pushing to the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+
+  tests:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+
+      # Run all jobs to completion (false), or cancel
+      # all jobs once the first one fails (true).
+      fail-fast: false
+
+      matrix:
+        os: [
+          "ubuntu-latest",
+          # "macos-latest",
```

Comment:
After bringing in #700, I think it will probably be safe to enable testing on macOS right away, if desired.

Reply:
I think it is equally important to add Windows to CI, to prevent basic errors like 601f9af9ca5da. Further efforts are made with #707 in this area.

On the other hand, I believe it will be crazy to run macOS- and Windows-tests on the _whole_ test matrix. I think it will be good enough to restrain them to Python 3.10 only, for example.